### PR TITLE
*: use host + port, the same as tidb and pd.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @c4pt0r  @qiuyesuifeng  @iamxy @BusyJay

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: src/bin/tikv-server.rs [hunk 1]
```diff
@@ -515,14 +523,12 @@ fn main() {
...
-                "set listening address",
-                "default is 127.0.0.1:20160");
+    opts.optopt("P", "port", "server listening port", "default is 20160");
     opts.optopt("",
-                "advertise-addr",
-                "set advertise listening address for client communication",
-                "127.0.0.1:20160, if not set, use addr instead.");
+                "advertise-port",
+                "advertise port for outer traffic",
+                "default is {port}");
```
Comment (BusyJay) [inline src/bin/tikv-server.rs]: "if not set, use port instead" seems better.

File: etc/config-template.toml [hunk 2]
```diff
@@ -6,10 +6,12 @@
...
-# set listening address.
-addr = "127.0.0.1:20160"
-# if not set, use addr instead. set advertise listening address for client communication.
-advertise-addr = ""
+# server host.
+host = "127.0.0.1"
+# server listening port.
+port = 20160
+# if not set, use port instead. set advertise port for client communication.
+advertise-port = ""
```
Comment (BusyJay) [inline etc/config-template.toml]: Advertise port seems useless. When will it be used?

Comment (siddontang) [inline etc/config-template.toml]: docker

File: src/bin/tikv-server.rs (hunk 1 above)
Comment (siddontang) [inline src/bin/tikv-server.rs]: the same like pd and etcd, but I should use ${port}

**Pull request closed without merging**